test/test_simple_decompression.py

```diff
@@ -38,6 +38,57 @@ def test_decompressobj(simple_compressed_file):
     assert data == uncompressed_data
 
 
+# `more_data_limit` allows testing `decompress(b'')` with and without a limit.
+@pytest.mark.parametrize('more_data_limit', [100, None])
+def test_decompressobj_with_output_buffer_limit(
+    simple_compressed_file, more_data_limit
+):
+    """
+    Test decompression with `output_buffer_limit` set.
+    """
+    with open(simple_compressed_file[0], 'rb') as f:
+        uncompressed_data = f.read()
+
+    with open(simple_compressed_file[1], 'rb') as f:
+        compressed_data = f.read()
+
+    o = brotlicffi.Decompressor()
+    assert o.can_accept_more_data()
+    small_limit = 100
+    result = o.decompress(compressed_data, output_buffer_limit=small_limit)
+    assert len(result) <= small_limit
+
+    # Ensure `output_buffer_limit` of zero works.
+    assert o.decompress(b'', output_buffer_limit=0) == b''
+
+    if o._unconsumed_data:
+        with pytest.raises(
+            brotlicffi.error,
+            match=(
+                r"brotli: decoder process called with data when "
+                r"'can_accept_more_data\(\)' is False"
+            ),
+        ):
+            o.decompress(b'additional data')
```

Comment:
Can you also add a check like this one? https://github.com/google/brotli/blob/f81d6bc7f07b860b9d6243e8d46ce4eda0db0855/python/tests/decompressor_test.py#L48

Reply:
@ngoldbaum thanks for the review!

The linked check confuses me because the test calls `decompressor.process` with `output_buffer_limit=10922` and then ensures that `len(decompressed_chunk) <= 32768 (MIN_OUTPUT_BUFFER_SIZE)`. Do you see any benefit of doing that?

In this test function, I added checks like `assert len(result) <= small_limit` and `assert len(more_output) <= more_data_limit` after successful calls.
Line 72 is expected to raise an error, so it has no such check.

Reply:
I was thinking of something along the lines of a test for the `input_data_len == 0` code path, where the implicit limit is 32 kb. Unless I’m misreading your code.

Looking at the test I linked to again, I agree that linking to the brotli source code was a red herring. Sorry! Still getting acquainted with both codebases…